Controllers/HostController.cs

```diff
@@ -139,6 +139,63 @@ public HttpResponseMessage CreateNewHost(Host host)
             }
             return Request.CreateResponse(HttpStatusCode.OK, $"NEW HOST CREATED SUCCESFULLY: {host.Name} {host.LastName}");
         }
+        // ===================================================================================================
+        [AllowAnonymous]
+        [HttpPost]
+        [Route("auth")]
+        public IHttpActionResult Authenticate(LoginRequest loginRequest)
+        {
+            if (loginRequest == null) return BadRequest("Complete the fields to login an host.");
+            Host host = new Host();
+            try
+            {
+                SqlConnection sqlConnection = new SqlConnection(DB_CONNECTION_STRING);
+                using (sqlConnection)
+                {
+                    SqlCommand sqlCommand = new SqlCommand(@"SELECT 
+                                                                    Hos_ID,
+                                                                    Hos_Name,
+                                                                    Hos_LastName,
+                                                                    Hos_Password,
+                                                                    Hos_Description,
+                                                                    Hos_Status 
+                                                                FROM Host
+                                                                WHERE Hos_Name = @Hos_Name 
+                                                                AND Hos_Password = @Hos_Password", sqlConnection);
+
+                    sqlCommand.Parameters.AddWithValue("@Hos_Name", loginRequest.Username);
+                    sqlCommand.Parameters.AddWithValue("@Hos_Password", loginRequest.Password);
+
+                    sqlConnection.Open();
+
+                    SqlDataReader sqlDataReader = sqlCommand.ExecuteReader();
+
+                    if (sqlDataReader.Read())
+                    {
+                        host.ID = sqlDataReader.GetInt32(0);
+                        host.Name = sqlDataReader.GetString(1);
+                        host.LastName = sqlDataReader.GetString(2);
+                        host.Password = sqlDataReader.GetString(3);
+                        host.Description = sqlDataReader.GetString(4);
+                        host.Status = sqlDataReader.GetString(5);
+
+                        var token = TokenGenerator.GenerateTokenJwt(loginRequest.Username);
+                        host.Token = token;
+                    }
+                    sqlConnection.Close();
```

Comment:
when a SQL connection is open inside a `using()` the connection is closed when the execution go out from the `using()` so is not necessary call a `Close()` inside a `using()`

Reference: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement

Reply:
Still there